src/main/kotlin/com/cognifide/gradle/aem/common/file/transfer/FileTransferSftp.kt

```diff
@@ -0,0 +1,121 @@
+package com.cognifide.gradle.aem.common.file.transfer
+
+import com.cognifide.gradle.aem.common.AemException
+import com.cognifide.gradle.aem.common.file.FileException
+import com.cognifide.gradle.aem.common.file.IoTransfer
+import java.io.File
+import java.io.IOException
+import net.schmizz.sshj.SSHClient
+import net.schmizz.sshj.sftp.*
+import org.apache.http.client.utils.URIBuilder
+
+class FileTransferSftp(
+    uploadUrl: String,
+    private val credentials: Credentials,
+    private val hostChecking: Boolean? = false,
+    private val ioTransfer: IoTransfer = IoTransfer()
+) : FileTransfer {
+
+    private val uploadUrl = uploadUrl.trimEnd('/')
+
+    init {
+        try {
+            connect { path ->
```

Comment:
why connection is being established when class is being constructed? is it necessary?
I wondering if connection should be active only when perfoming upload / download (encapsulated within single method)

Reply:
I will defer validation to the point when upload/download is started.